### DISCUSSION TITLE
Why is it we can’t control exposed patch.js easily?

### DISCUSSION THREAD

**Discussion by cristianvogel** [cables-gl/cables_docs#512] [Feature Requests] [answered]
Maybe I am missing something. Although I have been trying real hard to use the Function.op ... but my Svelte-Kit app in dev mode is simply not getting a valid CABLES.patch.config Object when I embed the exported patch in a component. I can't see what's happening because the CABLES instance doesn't exist until runtime and then I have to probe it in devtools to which is annoying.

What I want to do actually is programmatically control the Patch's GL.Main enable flag, so I can pause the patch from running if I have to - from my code, outside the patch.

I don't really understand why thats not simple, like exposed setters for the state of OPs once it's running locally.

**Comment by cristianvogel** [comment]
Next day... 

I managed to get the ` Ops.Cables.Function_v2` to fire by using the following check. 

```	function go() {
		
		if ( CABLES.hasOwnProperty('patch') ) {
			console.log('go: ', change);
			CABLES.patch.config.patchFunctionTrigger( );
		}
	}

**Comment by steam0r** [comment, marked as answer]
hey,

is this still related to svelte? there is plenty of options to do what i think you want to archive.

did you know you can set variables when initializing the patch?

```
CABLES.patch = new CABLES.Patch({
        ... // usual stuff
        variables:
        {
            "canColor":"#00ff00"
        }
    });
```

also you can pause and resume the patch using `CABLES.patch.pause();` and `CABLES.patch.resume();` 

or maybe you need to do your function-calls in any of these callbacks provided by cables?

```  
    function patchInitialized(patch) {
        // You can now access the patch object (patch), register variable watchers and so on
    }

    function patchFinishedLoading(patch) {
        // The patch is ready now, all assets have been loaded
    }
```

**Reply by cristianvogel** [reply]
OK, no I don't think its to do with Svelte, although it is compiling so could be someething to do with order of initialising things.. 

Are all the things we can do with `CABLES.patch` documented somewhere?

**Reply by steam0r** [reply]
where did you put your code that has that check?

documentation is here:

https://cables.gl/docs/4_export_embed/dev_embed_vars/dev_embed_vars
https://cables.gl/docs/4_export_embed/dev_embed_functions/dev_embed_functions
https://cables.gl/docs/4_export_embed/embedding/embedding

**Reply by cristianvogel** [reply]
Lordy - this answers the question to this thread. I am sorry - I had *not* seen that documentation about accessing the vars at all. This is perfect for today's hacking.  

Should I be deleting the [op.set.var] in the Cables patch when accessing the vars like this? 
My check is like this... 

If I don't do the check, then the patch.config doesn't exist

```
function patchInitialized() {
		// You can now access the patch object (CABLES.patch), register variable watchers and so on
		go(	)
	}

	function patchFinishedLoading() {
	
	}

	function go() {	
		if ( CABLES.hasOwnProperty('patch') ) {
			console.log('go: ', change);
			CABLES.patch.config.patchFunctionTrigger( );
		}
	}

**Reply by steam0r** [reply]
yeah that check i too early. try keeping the setters if you need default values when NOT setting them from the outside
